docs/handling-api-keys.md

```diff
@@ -0,0 +1,151 @@
+# Handling API keys
+
+A number of pgai functions call out to third-party APIs which require an API
+key for authentication. There are a few ways to pass your API key to these
+functions. This document lays out the different options and provides
+recommendations for which option to use.
+
+## pgai and API keys
+
+All pgai functions that require an API key take two optional parameters:
+`api_key_name`, and `api_key`. If neither parameter is provided, the function
+will fall back to a default name and attempt to extract an API key with that
+name. TODO: is this clear?
+
+We suggest that you generally prefer using `api_key_name`, as it is harder to
+misuse.
+
+### Using the `api_key` parameter
+
+When you use the `api_key` parameter, you pass the API key as a plain text
+value to the function. This is simple to use, but results in the API key being
+printed to the PostgreSQL logs. For this reason we DO NOT recommend this method.
+
+It is safe pass the `api_key` parameter as a bind variable:
+
+1. Set the API key as an environment variable in your shell:
+    ```bash
+    export MY_API_KEY="this-is-my-super-secret-api-key-dont-tell"
+    ```
+
+2. Connect to your database and set your api key as a [psql variable](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-VARIABLES):
+
+      ```bash
+      psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>" -v my_api_key=$MY_API_KEY
+      ```
+   Your API key is now available as a psql variable named `my_api_key` in your psql session.
+
+   You can also log into the database, then set `my_api_key` using the `\getenv` [metacommand](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-GETENV):
+
+      ```sql
+      \getenv my_api_key MY_API_KEY
+      ```
+
+3. Pass your API key to your parameterized query:
+    ```sql
+    SELECT *
+    FROM ai.openai_list_models(api_key=>$1)
+    ORDER BY created DESC
+    \bind :my_api_key
+    \g
+    ```
+
+### Using the `api_key_name` parameter
```

Comment:
I would break out the subsections here into their own sections and move the two recommended ones on top